python/cudf/cudf/core/index.py

```diff
@@ -1247,7 +1303,7 @@ def __new__(cls, values, **kwargs):
     def __initializer__(self, values, **kwargs):
```

Comment:
This is tangential to this issue, but what is `__initializer__`?

Reply:
I created instead of having an `__init__`, as other child class would trigger it. But I can add `__init__` to those child class and change `__initializer__` to `__init__`

Reply:
Gotcha, yeah relying on the normal `__init__` is preferable. If we want to have a method that does the work of a constructor, let's call it `_initialize()` or `_initializer()`.

[Double underscore methods](https://docs.python.org/3/reference/datamodel.html#special-method-names) are typically reserved for methods that are invoked using some special syntax.

Reply:
Sorry for asking for changes outside of the scope of this PR :(

Reply:
NP, I will take care of it.